lib/nsxv.py

```diff
@@ -238,12 +242,29 @@ def __str__(self):
           destination_address = nacaddr.ExcludeAddrs(
               destination_address,
               destination_address_exclude)
-        if not destination_address:
-          logging.warn(self.NO_AF_LOG_ADDR.substitute(term=self.term.name,
-                                                      direction='destination',
-                                                      af=self.filter_type))
+
+        if destination_address:
+          if af == 4:
+            dest_ipv4_address = True
+          else:
+            dest_ipv6_address = True
         destination_addr.extend(destination_address)
 
+    # Drop the mismatch IP if source and destination address are not valid
+    # for mixed filter
+    if self.filter_type == 'mixed':
+      if source_addr and destination_addr:
+        if source_ipv4_address and not dest_ipv4_address:
+          source_addr = self._DropMismatchIpAddress(source_addr, 'source', 4)
+        elif source_ipv6_address and not dest_ipv6_address:
+          source_addr = self._DropMismatchIpAddress(source_addr, 'source', 6)
+        elif dest_ipv4_address and not source_ipv4_address:
+          destination_addr = self._DropMismatchIpAddress(destination_addr,
+                                                 'destination', 4)
+        elif dest_ipv6_address and not source_ipv6_address:
+          destination_addr = self._DropMismatchIpAddress(destination_addr,
+                                                 'destination', 6)
```

Comment:
The whole idea of this was to eliminate rules which are not valid, a rule that is src ipv4 and dst to ipv6 is invalid and thus should be removed. If the result after filtering results in 0 addresses on one side the expectation is not to add in an any, that would open up a  huge hole. It should remove the rule in this case.